crates/fs/common/utils.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::Result;
+use serde::de::Visitor;
+use std::fmt;
+
+use crate::{error, FsError};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+pub(crate) struct ByteArrayVisitor<const N: usize>;
+
+#[cfg(test)]
+pub(crate) struct TestRng();
+
+//--------------------------------------------------------------------------------------------------
+// Implementations
+//--------------------------------------------------------------------------------------------------
+
+impl<'de, const N: usize> Visitor<'de> for ByteArrayVisitor<N> {
+    type Value = [u8; N];
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a byte array of length {}", N)
+    }
+
+    fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        let bytes: [u8; N] = v.try_into().map_err(E::custom)?;
+        Ok(bytes)
+    }
+}
+
+#[cfg(test)]
+impl crate::private::Rng for TestRng {
```

Comment:
This looks useful. A more flexible Rng for proptests. I don't think we need `&mut self`.
The only worry I have is shrinking on seeds. Is that a useful property to test?

Reply:
Well, at worst it makes shrinking slow, but I agree, shrinking a seed is usually useless. However, shrinking can be turned off only some test inputs easily.

Reply:
> I don't think we need it to be `&mut self`.

Well, I think we do, otherwise we wouldn't be able to do what I described in the comment above with proptest's `TestRng`, because it needs `&mut self`:
https://docs.rs/proptest/latest/proptest/prelude/trait.RngCore.html#tymethod.fill_bytes

Reply:
To be clear: We'd need `&mut self` *for doing that "fun idea" with proptest*.
I'd say we go ahead without &mut right now!